src/verilog/verilog_synthesis.cpp

```diff
@@ -1389,6 +1390,57 @@ void verilog_synthesist::synth_initial(
 
 /*******************************************************************\
 
+Function: verilog_synthesist::make_supply_value
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+exprt verilog_synthesist::make_supply_value(
+  const irep_idt &decl_class,
+  const typet &type)
+{
+  if(type.id() == ID_array)
+  {
+    auto element =
+      make_supply_value(decl_class, to_array_type(type).element_type());
+    return array_of_exprt(element, to_array_type(type));
```

Comment:
Could use `if(auto array_type = type_try_dynamic_cast<array_typet>(type))` and then just use `array_type->element_type()` and `*array_type` to avoid the repeated `to_array_type`.